zebra-rpc/src/methods/tests/vectors.rs

```diff
@@ -141,3 +141,24 @@ async fn rpc_getbestblockhash() {
 
     mempool.expect_no_requests().await;
 }
+
+#[tokio::test]
+async fn rpc_getrawmempool() {
+    zebra_test::init();
+
+    let mut mempool: MockService<_, _, _, BoxError> = MockService::build().for_unit_tests();
+    let mut state: MockService<_, _, _, BoxError> = MockService::build().for_unit_tests();
+
+    // Init RPC
+    let _rpc = RpcImpl {
+        app_version: "Zebra version test".to_string(),
+        mempool: Buffer::new(mempool.clone(), 1),
+        state: Buffer::new(state.clone(), 1),
+    };
+
+    // Hangs
+    //let raw_mempool = rpc.get_raw_mempool().await.unwrap();
```

Comment:
This line seems to be hanging in the ready() of the method: https://github.com/ZcashFoundation/zebra/blob/7ffa68d55680c7fccae333fa8e6db084b3163621/zebra-rpc/src/methods.rs#L305 and cant move forward.

I am unsure if it is caused because we dont have a mempool. Actually, for a complete test we should have some transactions in the mempool. All the tests we have now that have a mempool are in zebrad, for example but not the only one available:

https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/mempool/tests/vector.rs#L694

I am wondering if anyone has any idea on what we should do here. Should we move one of those setup functions into maybe `node-services` crate or do we have a simpler alternative to have a mempool in the `zebra-rpc` crate?

Reply:
This happens because it's waiting for a response from the mempool service, and the `MockService` only responds if we tell it to.

We'd have to do something like:

```rust
    let call_task = tokio::spawn(rpc.get_raw_mempool());

    mempool
        .expect_request(mempool::Request::TransactionIds)
        .await
        .respond(mempool::Response::TransactionIds(transaction_ids));

    // The mempool service should only get one call
    // so we should check there aren't any others.
    mempool.expect_no_requests().await;
```

I suggest making this test a property test, so that we can generate random `transaction_ids`.

Reply:
This is simpler and a lot better than what i had in mind. I actually used this before but i forgot how it works, sorry about that and thanks!